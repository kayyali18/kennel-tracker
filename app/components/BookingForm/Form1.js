import styled from 'styled-components'

const Form1 = styled.form`
  background-color: beige;
  display: grid;
  grid-gap: 4rem;
  grid-template-areas: 'input .';
  height: 100%;

  position: absolute;
  overflow: scroll;
  width: 100%;
  input {
    border: 2px solid black;
    position: relative;
    height: 50px;
  }

  input:nth-child(odd) {
    top: 2rem;
    left: 4rem;
  }
  input:nth-child(even) {
    top: 6rem;
    right: 4rem;
  }
`
export default Form1

// make them checkboxes so when checked they can move to the next
