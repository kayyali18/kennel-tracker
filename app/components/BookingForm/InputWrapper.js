import styled from 'styled-components'

const InputWrapper = styled.div`
  display: flex;
  flex-direction: row-reverse;
  border-bottom: 1px solid rgba(#fffce8, 0.1);
  padding-bottom: 3px;
  margin-bottom: 5px;

  input {
    outline: none;
    border: 0;
    color: #fffce8;
    background: transparent;
    font-family: 'Lato';
    flex: 1 0 auto;
    font-size: 14px;
    font-weight: 300;
  }

  input[type='text'] {
    ~ label {
      &::before {
        content: '/f007';
      }
    }
  }

  label:focus {
    opacity: 1;
  }

  label {
    font-family: 'Font Awesome';
    font-size: 13px;
    opacity: 0.1;
    transition: opacity 0.5s ease;
  }

  label::before {
    align-items: center;
    color: #fffce8;
    display: flex;
    height: 30px;
    justify-content: center;
    width: 30px;
  }
`
export default InputWrapper
