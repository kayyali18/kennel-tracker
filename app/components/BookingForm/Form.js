import React from 'react'

// import Form 1
import Form1 from './Form1'

function Form() {
  let form = 0

  form = (
    <Form1>
      <input placeholder="Name" type="text" />
      <input placeholder="Address" type="text" />
      <input placeholder="Cellphone" type="number" />
      <input placeholder="Homephone" type="number" />
      <input placeholder="E-mail" type="email" />
      <input placeholder="Emergency Contact Name" type="text" />
      <input placeholder="Emergency Contact #" type="number" />
    </Form1>
  )

  return <div>{form}</div>
}

export default Form
