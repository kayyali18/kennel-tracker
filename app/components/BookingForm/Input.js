import React from 'react'
import PropTypes from 'prop-types'
import InputWrapper from './InputWrapper'

function Input(props) {
  return (
    <InputWrapper>
      <input
        onChange={props.handleChange}
        id={props.id}
        autoComplete="true"
        required
        type={props.type}
        placeholder={props.placeholder}
      />
    </InputWrapper>
  )
}

export default Input

Input.propTypes = {
  name: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  type: PropTypes.string,
  placeholder: PropTypes.string,
}
