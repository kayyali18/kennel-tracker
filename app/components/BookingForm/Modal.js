import React from 'react'
import PropTypes from 'prop-types'

import Input from './Input'
import ModalWrapper from './ModalWrapper'

function Modal(props) {
  return (
    <ModalWrapper>
      <form onSubmit={props.onSubmit} className="ModalForm">
        <Input id="name" type="text" placeholder="John Johnson" />
        <Input
          id="username"
          type="email"
          placeholder="JohnsonAndJohnson@gmail.com"
        />
        <Input id="password" type="password" placeholder="password" />
        <button type="submit">
          Log in <i className="fa fa-fw fa-chevron-right" />
        </button>
      </form>
    </ModalWrapper>
  )
}

export default Modal

Modal.propTypes = {
  onSubmit: PropTypes.func,
}
