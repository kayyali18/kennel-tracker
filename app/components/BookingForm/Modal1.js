import React from 'react'
import PropTypes from 'prop-types'

import Input from './Input'
import ModalWrapper from './ModalWrapper'

class Modal extends React.PureComponent {
  constructor(props) {
    super(props)
    this.state = {}
  }

  handleChange = e => {
    this.setState({
      [e.target.id]: e.target.value,
    })
  }

  handleSubmit = () => {
    this.props.onSubmit({ ...this.state, stage: 1 })
  }

  render() {
    return (
      <ModalWrapper>
        <h1> Owner Info</h1>

        <form
          onSubmit={e => {
            e.preventDefault()
            this.handleSubmit()
          }}
          className="ModalForm"
        >
          <Input
            handleChange={this.handleChange}
            id="name"
            type="text"
            placeholder="Name"
          />
          <Input
            handleChange={this.handleChange}
            id="address"
            type="text"
            placeholder="Address"
          />
          <Input
            handleChange={this.handleChange}
            id="home-phone"
            type="tel"
            placeholder="Home Telephone"
          />
          <Input
            handleChange={this.handleChange}
            id="cell-phone"
            type="tel"
            pattern="[0-9]{3}-[0-9]{3}-[0-9]{4}"
            placeholder="Cellphone"
          />
          <Input
            handleChange={this.handleChange}
            id="email"
            type="email"
            placeholder="E-mail"
          />
          <Input
            handleChange={this.handleChange}
            id="e-phone"
            type="tel"
            placeholder="Emergency Contact #"
          />
          <button type="submit">
            Log in <i className="fa fa-fw fa-chevron-right" />
          </button>
        </form>
      </ModalWrapper>
    )
  }
}

export default Modal

Modal.propTypes = {
  onSubmit: PropTypes.func,
}
