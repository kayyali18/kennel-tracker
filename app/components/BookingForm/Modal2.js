import React from 'react'
import PropTypes from 'prop-types'

import Input from './Input'
import ModalWrapper from './ModalWrapper'

class Modal extends React.PureComponent {
  constructor(props) {
    super(props)
    this.state = {}
  }

  handleChange = e => {
    this.setState({
      [e.target.id]: e.target.value,
    })
  }

  handleSubmit = () => {
    this.props.onSubmit({ ...this.state })
  }

  render() {
    return (
      <ModalWrapper>
        <h1> Pet Info</h1>

        <form
          onSubmit={e => {
            e.preventDefault()
            this.handleSubmit()
          }}
          className="ModalForm"
        >
          <Input
            handleChange={this.handleChange}
            id="name"
            type="text"
            placeholder="Pet Name"
          />
          <Input
            handleChange={this.handleChange}
            id="species"
            type="text"
            placeholder="Species"
          />
          <Input
            handleChange={this.handleChange}
            id="breed"
            type="text"
            placeholder="Pet Breed"
          />
          <Input
            handleChange={this.handleChange}
            id="color"
            type="text"
            placeholder="Pet Color"
          />
          <Input
            handleChange={this.handleChange}
            id="dob"
            type="date"
            data-placeholder="Date of Birth"
          />
          <select id="spayed_neutered" required onChange={this.handleChange}>
            <option value="">-- Please Select an Option --</option>
            <option value>Spayed/Neutered</option>
            <option value={false}>N/A</option>
          </select>
          <Input
            handleChange={this.handleChange}
            id="medications"
            type="text"
            placeholder="Medications"
          />
          <Input
            handleChange={this.handleChange}
            id="feeding_instructions"
            type="text"
            placeholder="Feeding Instructions"
          />
          <Input
            handleChange={this.handleChange}
            id="shots"
            type="text"
            placeholder="Shots"
          />

          <button type="submit">
            Next <i className="fa fa-fw fa-chevron-right" />
          </button>
        </form>
      </ModalWrapper>
    )
  }
}

export default Modal

Modal.propTypes = {
  onSubmit: PropTypes.func,
}
