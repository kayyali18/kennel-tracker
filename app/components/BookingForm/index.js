import React from 'react'
import PropTypes from 'prop-types'

import Modal1 from './Modal1'
import Modal2 from './Modal2'
import Modal3 from './Modal3'
import Wrapper from './Wrapper'

/* eslint-disable react */
function BookingForm(props) {
  let form1
  let form2
  let form3
  if (!props.mounted) {
    form1 = <Modal1 onSubmit={props.handleSubmit} />
    form2 = <Modal2 onSubmit={props.handleSubmit} />
    form3 = <Modal3 onSubmit={props.handleSubmit} />
  }

  if (props.stage == 1) return <Wrapper>{form1}</Wrapper>
  if (props.stage == 2) return <Wrapper>{form2}</Wrapper>
  if (props.stage == 3) return <Wrapper>{form3}</Wrapper>
  return <Wrapper>Thank You</Wrapper>
}

export default BookingForm

BookingForm.propTypes = {
  handleSubmit: PropTypes.func,
  mounted: PropTypes.bool,
  stage: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
}
