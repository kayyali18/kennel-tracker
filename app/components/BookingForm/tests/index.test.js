import React from 'react'
import renderer from 'react-test-renderer'

import BookingForm from '../index'

describe('<BookingForm />', () => {
  it('should match snapshot', () => {
    const renderedComponent = renderer.create(<BookingForm />).toJSON()
    expect(renderedComponent).toMatchSnapshot()
  })
})
