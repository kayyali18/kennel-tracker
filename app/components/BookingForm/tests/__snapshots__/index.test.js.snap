// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookingForm /> should match snapshot 1`] = `
<article
  className="Wrapper-sc-1w8d0gj-0 dGnIKR"
>
  <div>
    <form
      className="Form1-sc-1pwvulr-0 jOSPTR"
    >
      <input
        placeholder="Name"
        type="text"
      />
      <input
        placeholder="Address"
        type="text"
      />
      <input
        placeholder="Cellphone"
        type="number"
      />
      <input
        placeholder="Homephone"
        type="number"
      />
      <input
        placeholder="E-mail"
        type="email"
      />
      <input
        placeholder="Emergency Contact Name"
        type="text"
      />
      <input
        placeholder="Emergency Contact #"
        type="number"
      />
    </form>
  </div>
</article>
`;
